middleware/http/client.go

```diff
@@ -50,45 +59,49 @@ func TransportOptions(options ...TransportOption) ClientOption {
 
 // NewClient returns an HTTP Client adding Zipkin instrumentation around an
 // embedded standard Go http.Client.
-func NewClient(tracer *zipkin.Tracer, client *http.Client, options ...ClientOption) (*Client, error) {
+func NewClient(tracer *zipkin.Tracer, options ...ClientOption) (*Client, error) {
 	if tracer == nil {
 		return nil, ErrValidTracerRequired
 	}
 
-	if client == nil {
-		client = &http.Client{}
-	}
-
-	c := &Client{tracer: tracer, Client: client}
+	c := &Client{tracer: tracer, Client: &http.Client{}}
 	for _, option := range options {
 		option(c)
 	}
 
 	c.transportOptions = append(
 		c.transportOptions,
 		// the following Client settings override provided transport settings.
-		RoundTripper(client.Transport),
+		RoundTripper(c.Client.Transport),
 		TransportTrace(c.httpTrace),
 	)
 	transport, err := NewTransport(tracer, c.transportOptions...)
 	if err != nil {
 		return nil, err
 	}
-	client.Transport = transport
+	c.Client.Transport = transport
 
 	return c, nil
 }
 
-// DoWithTrace wraps http.Client's Do with tracing using an application span.
-func (c *Client) DoWithTrace(req *http.Request, name string) (res *http.Response, err error) {
-	appSpan := c.tracer.StartSpan(name, zipkin.Kind(model.Client))
+// DoWithAppSpan wraps http.Client's Do with tracing using an application span.
```

Comment:
what does "application span" mean? At first I assumed it was about the span created, but then noticed the span created has kind.client

Reply:
yes that's a left over error... I will remove the zipkin.Kind